spec/project_spec.rb

```diff
@@ -3939,4 +4138,173 @@ def callback(_args); end
+              "The user 'user1' is not bucketed into any of the experiments on the feature 'multi_variate_feature'.",
+              "Feature flag 'multi_variate_feature' is not used in a rollout."
+            ],
+            decision_event_dispatched: true
+          )
+        allow(custom_project_instance.event_dispatcher).to receive(:dispatch_event).with(instance_of(Optimizely::Event))
+        user_context = custom_project_instance.create_user_context('user1')
+        decision = custom_project_instance.decide(user_context, 'multi_variate_feature')
+        expect(decision.as_json).to include(
+          flag_key: 'multi_variate_feature',
+          enabled: false,
+          reasons: [
+            "User 'user1' is not in the forced variation map.",
+            "User 'user1' does not meet the conditions to be in experiment 'test_experiment_multivariate'.",
+            "The user 'user1' is not bucketed into any of the experiments on the feature 'multi_variate_feature'.",
+            "Feature flag 'multi_variate_feature' is not used in a rollout."
+          ],
+          rule_key: nil,
+          user_context: {attributes: {}, user_id: 'user1'},
+          variables: {'first_letter' => 'H', 'rest_of_name' => 'arry'},
+          variation_key: nil
+        )
+      end
+
+      it 'should not include reasons when the option is not set in default_decide_options' do
+        custom_project_instance = Optimizely::Project.new(config_body_JSON, nil, spy_logger, error_handler)
+        expect(custom_project_instance.notification_center).to receive(:send_notifications)
+          .once.with(Optimizely::NotificationCenter::NOTIFICATION_TYPES[:LOG_EVENT], any_args)
+        expect(custom_project_instance.notification_center).to receive(:send_notifications)
+          .once.with(
+            Optimizely::NotificationCenter::NOTIFICATION_TYPES[:DECISION],
+            'flag',
+            'user1',
+            {},
+            flag_key: 'multi_variate_feature',
+            enabled: false,
+            variables: {'first_letter' => 'H', 'rest_of_name' => 'arry'},
+            variation_key: nil,
+            rule_key: nil,
+            reasons: [],
+            decision_event_dispatched: true
+          )
+        allow(custom_project_instance.event_dispatcher).to receive(:dispatch_event).with(instance_of(Optimizely::Event))
+        user_context = custom_project_instance.create_user_context('user1')
+        decision = custom_project_instance.decide(user_context, 'multi_variate_feature')
+        expect(decision.as_json).to include(
+          flag_key: 'multi_variate_feature',
+          enabled: false,
+          reasons: [],
+          rule_key: nil,
+          user_context: {attributes: {}, user_id: 'user1'},
+          variables: {'first_letter' => 'H', 'rest_of_name' => 'arry'},
+          variation_key: nil
+        )
+      end
+    end
+
+    describe 'DISABLE_DECISION_EVENT' do
```

Comment:
This default_decide_option testing misses a test with "ENABLED_FLAGS_ONLY", with which we could have detected the bug in "decide_for_keys" not checking default_decide_options :)